janusgraph-core/src/main/java/org/janusgraph/diskstorage/log/kcvs/ExternalCachePersistor.java

```diff
@@ -38,7 +40,9 @@ public ExternalCachePersistor(KCVSCache kcvs, CacheTransaction tx) {
     @Override
     public void add(StaticBuffer key, Entry cell) {
         try {
-            kcvs.mutateEntries(key, Lists.newArrayList(cell), KCVSCache.NO_DELETIONS,tx);
+            List<Entry> list = new ArrayList<>(1);
+            list.add(cell);
+            kcvs.mutateEntries(key, list, KCVSCache.NO_DELETIONS,tx);
```

Comment:
From guava documentation `newArrayList` 

> Creates a mutable ArrayList instance containing the given elements.

From java docs `Arrays.asList`

> Returns a fixed-size list backed by the specified array

As an array cannot change its size, it can only replace it's elements by other elements but no addition or deletion are supported. So, `new ArrayList` and `Arrays.asList` are different lists. If some code will try to remove or add something into that list, we will get `UnsupportedOperationException`.

Reply:
Good to know! Thank you for pointing that out